#! /usr/bin/env python


# Initialize ros and import parameter generator
PACKAGE='robot_shared_control'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Add parameters
#       Name     Type    Level Description     	Default 	Min   	Max
gen.add("filteredForceGain",double_t,0,"Filtering weight of force torque sensor",0.9,0,1)
gen.add("velocityLimit",double_t,0,"Velocity limit [m/s]",0.3,0.3,1)
gen.add("duration",double_t,0,"Duration [s]",60,10,180)
gen.add("kFalcon", double_t,0,"Falcon stiffness parameter",0,0,1000)
gen.add("dFalcon", double_t,0,"Falcon damping parameter",0,0,100)

exit(gen.generate(PACKAGE, "robot_shared_control", "falconControl_params"))